lib/internal/abort_controller.js

```diff
@@ -357,6 +362,24 @@ function transferableAbortController() {
   return AbortController[kMakeTransferable]();
 }
 
+/**
+ * @param {AbortSignal} signal
+ * @param {any} resource
+ * @returns {Promise<void>}
+ */
+function aborted(signal, resource = null) {
+  try {
+    abortSignalValidator(signal, 'signal');
+  } catch (err) {
+    return PromiseReject(err);
+  }
+  if (signal.aborted)
+    return PromiseResolve();
+  const abortPromise = createDeferredPromise();
+  signal.addEventListener('abort', abortPromise.resolve, { [kWeakHandler]: resource, once: true });
```

Comment:
yes same as

Reply:
Okay, then my next question is: why add another function to do the same thing? :)

Reply:
basically as far as I understand from the https://github.com/nodejs/node/issues/37220#issue-801068533 description here we want to reduce the code required for cleanup by including weak listeners

Reply:
basically this exposes

```js
require('events').once(signal, 'abort', { [kWeakHandler]: /*something to cleanup*/ }).then(/* ... */)
``` 
to users

Reply:
> For my understanding, is this line basically identical to events.once?

No, since with `aborted` the abort listener is cleaned up automatically using weak event listeners. This means you don't have to listen to whatever means the resource is no longer relevant and clean up the abort listener up. Which was a pain point in Node.js core cancellation support.